File: Tai_lieu/Code/sum.py

v=[]
def init():
	for i in range(1,10):
		val=0
		for j in range (31):
			val=val*10+i
			v.append(val)


init()
n=int(input())
v.sort()
res=-1
for i in v:
	if (i%n==0):
		res=i/n 
		break
print(int(res))

File: Tai_lieu/Code/sum.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long

vector <int> v;
void init()
{
	for (int i=1;i<9;i++)
	{
		int k=1,val=0;
		while (k--) val=val*10+i,v.push_back(val);
	}
}

signed main()
{
	int n; cin >> n;
	init();
	sort(v.begin(),v.end());
	for (int i=0;i<v.size();i++)
	{
		if (v[i]%n==0) return cout << v[i]/n,0;
	}
	cout << -1;
	return 0;
}